at-date-api/src/main/kotlin/com/amerharb/atdate/app.kt

```diff
@@ -0,0 +1,115 @@
+package com.amerharb.atdate
+
+import io.ktor.http.*
+import io.ktor.server.application.*
+import io.ktor.server.engine.embeddedServer
+import io.ktor.server.netty.Netty
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import java.util.*
+
+const val PORT = 8000
+
+fun main() {
+    embeddedServer(Netty, port = PORT, module = Application::module).start(wait = true)
+}
+
+fun Application.module() {
+    routing {
+        get("/") {
+            call.respondText("Hello, world!")
+        }
+
+        post("/encode/{notation}") {
+            val notation = call.parameters["notation"]
+            if (notation == null) {
+                call.respondText("Notation is null")
+                return@post
+            }
+            // if encoding throw error respond with error
+            try {
+                val ad = encode(notation)
+
+                val res = """
+                    hex: ${ad.getHex()}
+                    base64: ${ad.getBase64()}
+            """.trimIndent()
+                call.respondText(res)
+            } catch (e: Exception) {
+                call.respondText(
+                    text = e.message ?: "Error",
+                    contentType = ContentType.Text.Plain,
+                    status = HttpStatusCode.InternalServerError,
+                )
+                return@post
+            }
+        }
+
+        post("/encode/{notation}/hex") {
+            val notation = call.parameters["notation"]
+            if (notation == null) {
+                call.respondText("Notation is null")
+                return@post
+            }
+            val ad = encode(notation)
+            call.respondText(ad.getHex())
+        }
+
+        post("/encode/{notation}/base64") {
+            val notation = call.parameters["notation"]
+            if (notation == null) {
+                call.respondText("Notation is null")
+                return@post
+            }
+            val ad = encode(notation)
```

Comment:
i would consider doing the encode/decode to be called asynchronously.
        val ad = withContext(Dispatchers.Default) { decode(bArray) }

Reply:
backend development is not similar to Android. With backend we dont mind blocking the caller thread. Actually you want to block it in 90% of the cases to send back a results

Reply:
Its actually true that in backend , we often perform operations synchronously and block the caller thread, asynchronous operations can still be beneficial in certain situations, especially for long-running or heavy computational tasks, i havent checked how the decode/encode function heaviness yet .. since the functions are so big. But i think making calls asynchronously is not necessary here .  P.S @amerharb is using Ktor, and  a lot of the concurrency management is handled automatically, meaning each request is handled in its own coroutine, keeping the server responsive to other incoming requests.

Reply:
encode and decode is pure arithmetic function, there is no blocking resource  during its call, it is better to stay as sync